### DISCUSSION TITLE
Développement d’un site de vente de  Jouets en ligne

### DISCUSSION THREAD

**Discussion by Mari12em** [community/community#145579] [Codespaces]
### Select Topic Area

Show & Tell

### Body

Nous souhaitons faire une solution complète de vente de jouets en ligne qui offre les
fonctions standard de fonctionnement d’un magasin de vente de jouets à savoir
chercher un jouet, commander et payer un jouet ...
Ce logiciel dispose d’une interface graphique Web simpliste. C’est à dire qu’on
demande des pages très simples et minimaliste (il ne s’agit pas de faire du html
complexe). Ce logiciel doit pouvoir aussi fonctionner en mode Client/Serveur sans
changer le code métier.
En réalité il existe trois profils de personnes qui peuvent utiliser cette Société de
Location en Ligne :
- Client
- Gestionnaire du Magasin de vente en ligne
- Administrateur Site
Rôle du Client :
- S’inscrire dans le système pour bénéficier d’un code Client en fournissant nom,
prénom, email et autres infos...
- Changer son mot de passe et son profil
- S’authentifier dans le système
- Chercher un jouet selon un mot clef et vérifier sa disponibilité
- Afficher les caractéristiques détaillées du jouet
- Choisir le jouet le mettre dans son panier
- Afficher sa facture et payer
Rôle du Gestionnaire
- S’authentifie dans le système
- Faire un inventaire complet de tous les jouets
- Ajouter un jouet
- Afficher la facture du client a partir de son code et si possible l’imprimer
- Supprimer un jouet
- Afficher si un jouet est disponible ou pas
- Poster un message pour afficher une promotion a tous les clients
Rôle de l’administrateur
- Supprimer un client du système
- Créer et modifier le profil du Gestionnaire

- 2 -

2-Objectif du Projet :
- Développer une solution complète avec les outils et les connaissances dont vous
disposez. Vous devez réaliser une conception en UML et développer touts les
composants Java correspondants a cette conception.
- Pour la programmation Java il sera essentiel de pratiquer la programmation
défensive c’est a dire utiliser les exceptions, prendre en compte tous les cas
particuliers et faire les test unitaires et éventuellement les assertions.
- Du point de vue développement, le projet inclut la mise en place de tous les
éléments suivants :
- Pages html simpliste
- L’ensemble des composants métier Java
- L’ensemble des servlets
- La conception de la Base de Données
- L’intégration et la mise en marche de l’ensemble

**Comment by Mari12em** [comment]
Did you help me create my project I am a beginner in this field

**Comment by shinybrightstar** [comment]
Hi @Mari12em , thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
